### ISSUE TITLE
test: rework coverage handling

### ISSUE THREAD

**Issue by heyitsanthony** [coreos/etcd#5930]
We want to pay more attention to test coverage; right now there is only coverage testing for unit tests and it's generally ignored. Ideally `./test` would run all tests with coverage taking enabled and then merge the results into one coverage file.

As preliminary investigation, I tried coverage testing on integration tests with

```
go test -coverpkg=`go list -f '{{ .Deps }}' | tr ' ' '\n' | grep etcd | tr '\n' ','`  -coverprofile cover.out -v
```

which seems to work OK.

Getting coverage out of e2e tests might be possible with some gross hacks around build tags and `go test -c` to produce etcd and etcdctl binaries that will write out coverage information on exit.

**Comment by fanminshi** [comment]
code coverage for unit and integration is done via https://github.com/coreos/etcd/pull/7067

Need to do coverage for e2e tests.

**Comment by fanminshi** [comment]
closed via #7067 #7254 #7387
